lib/host/gen_helpers.js

```diff
@@ -60,14 +74,18 @@ module.exports = (state = {}, adapter) => {
      */
     emitSelfAsync: async (eventName, ...eventArgs) => {
       return new Promise((resolve, reject) => {
-        setTimeout(() => {
+        const t = setTimeout(() => {
+          timeoutObject.t = null
           logHelperCall('emitSelfAsync: %s', eventName)
 
           state.ev
             .emit(`self:${eventName}`, ...eventArgs)
             .then(resolve)
```

Comment:
Maybe cleanup from pendingOrderSubmitCancelTimeouts in a `.finally` function?

Reply:
You mean setting it to `null`? I will rename the var.